packages/mui-base/src/SliderUnstyled/SliderUnstyled.js

```diff
@@ -626,32 +92,64 @@ const SliderUnstyled = React.forwardRef(function SliderUnstyled(props, ref) {
     valueLabelFormat,
   };
 
+  const {
+    axisProps,
+    getRootProps,
+    getHiddenInputProps,
+    open,
+    active,
+    axis,
+    range,
+    focusVisible,
+    handleMouseOver,
+    handleMouseLeave,
+    dragging,
+    marks,
+    values,
+    trackOffset,
+    trackLeap,
+  } = useSlider({ ...ownerState, ref });
+
+  ownerState.marked = marks.length > 0 && marks.some((mark) => mark.label);
+  ownerState.dragging = dragging;
+
+  const Root = component ?? components.Root ?? 'span';
+  const rootProps = appendOwnerState(Root, { ...other, ...componentsProps.root }, ownerState);
+
+  const Rail = components.Rail ?? 'span';
+  const railProps = appendOwnerState(Rail, componentsProps.rail, ownerState);
+
+  const Track = components.Track ?? 'span';
+  const trackProps = appendOwnerState(Track, componentsProps.track, ownerState);
+  const trackStyle = {
+    ...axisProps[axis].offset(trackOffset),
+    ...axisProps[axis].leap(trackLeap),
+  };
+
+  const Thumb = components.Thumb ?? 'span';
+  const thumbProps = appendOwnerState(Thumb, componentsProps.thumb, ownerState);
+
+  const ValueLabel = components.ValueLabel ?? SliderValueLabelUnstyled;
+  const valueLabelProps = appendOwnerState(ValueLabel, componentsProps.valueLabel, ownerState);
+
+  const Mark = components.Mark ?? 'span';
+  const markProps = appendOwnerState(Mark, componentsProps.mark, ownerState);
+
+  const MarkLabel = components.MarkLabel ?? 'span';
+  const markLabelProps = appendOwnerState(MarkLabel, componentsProps.markLabel, ownerState);
+
   const classes = useUtilityClasses(ownerState);
 
   return (
     <Root
-      ref={handleRef}
-      onMouseDown={handleMouseDown}
+      {...getRootProps()}
```

Comment:
Pass in the external event handlers, so that ours `onMouseDown` is not overridden by the caller accidentally.

Reply:
It is handled in the hook itself, it was removed from here.

Reply:
I meant that if someone adds an `onMouseDown` prop on `SliderUnstyled`, it'll shadow our internal handler defined in `useSlider`. To prevent this, pass in `other` to `getRootProps`. 
Then, ideally, the `createHandleMouseDown` function in `useSlider` should check if `defaultPrevented` was set and skip its execution if it was.

Reply:
It is destructured from the props, and the hook already handles this. Let me refactor to explicitly send it in the hook.

Reply:
Addressed with - https://github.com/mui-org/material-ui/pull/30094/commits/afac9ca0e7baa71c913475d512476fa0cd4fe204